packages/eslint-plugin-baseui/src/deprecated-theme-api.js

```diff
@@ -0,0 +1,434 @@
+  },
+  font1350: {
+    concern: 'typography',
+    replacement: 'DisplayMedium',
+  },
+  font1450: {
+    concern: 'typography',
+    replacement: 'DisplayLarge',
+  },
+};
+
+module.exports = {
+  meta: {
+    fixable: 'code',
+    messages: {
+      [MESSAGES.deprecateThemeProperty.id]:
+        MESSAGES.deprecateThemeProperty.message,
+      [MESSAGES.replaceThemeProperty.id]: MESSAGES.replaceThemeProperty.message,
+    },
+  },
+  create(context) {
+    return {
+      Identifier(node) {
+        const identifier = node.name;
+
+        // We use `hasOwnProperty`, otherwise we might get false positives on the prototype chain.
+        if (
+          Object.prototype.hasOwnProperty.call(
+            deprecatedThemeProperties,
+            identifier,
+          )
+        ) {
+          // We have found an identifier that matches one of our deprecated theme properties.
+          const sourceCode = context.getSourceCode();
+          const scope = context.getScope(node);
+          const data = {
+            old: identifier,
+          };
+          let fix = null;
+          let messageId = MESSAGES.deprecateThemeProperty.id;
+
+          // If there is a possible replacement for this deprecated property,
+          // add a fix to replace with a new semantic property.
+          // TODO: use `suggest` instead, currently not supported in VSCode extension:
+          // https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions
+          // https://github.com/microsoft/vscode-eslint/issues/806
+          if (deprecatedThemeProperties[identifier].replacement) {
+            messageId = MESSAGES.replaceThemeProperty.id;
+            data.new = deprecatedThemeProperties[identifier].replacement;
+            fix = function(fixer) {
+              return fixer.replaceText(
+                node,
+                deprecatedThemeProperties[identifier].replacement,
+              );
+            };
+          }
+
+          // Check if we are setting a property in createTheme:overrides
+          // Ex: createTheme({}, {datepickerBackground: "pink"});
+          if (
+            context
+              .getAncestors()
+              .some(
+                node =>
+                  node.type === 'CallExpression' &&
+                  node.callee.name === 'createTheme',
+              ) &&
+            node.parent.type === 'Property' &&
+            node === node.parent.key
+          ) {
+            // A deprecated property is being set in createTheme.
+            // Assuming a violation.
+            context.report({
+              node,
+              messageId,
+              data,
+              fix,
+            });
+            return;
+          }
+
+          // Exit early if we are not in a function.
+          // We don't have access to the theme outside of functions.
+          // Ex: styled, withStyle, overrides, function component (useStyletron)
+          if (scope.type !== 'function') {
```

Comment:
what about a class method?

Reply:
How would you access the theme in a class method?

Reply:
render and a styled component created there, maybe not that common though

Reply:
I think the scope will always be a function because the node's parent scope is the body of the function passed to styled.

```js
class Foo extends React.Component {
  render() {
    const Boo = styled('div', ({$theme}) => {
      return {
        backgroundColor: $theme.colors.background // parent scope is arrow function
      }
    });
    return <Boo />;
  }
}
```

Reply:
Ah good point 👍